app/controllers/ontologies_redirection_controller.rb

```diff
@@ -47,32 +49,45 @@ def redirect_ontology
       
     private
       
-    def generate_htaccess_content(ontology_portal_url, ontology_uri)
-        ontology_rule = nil
-        if ontology_uri
-            ontology_uri += '/' unless ontology_uri.end_with?('/')
-            ontology_rule = "RewriteRule ^#{URI.parse(ontology_uri).path[1..-1]}?$ #{ontology_portal_url} [R=301,L]"
-        end
-
+    def generate_htaccess_content(ontology_portal_url, ontology_uri, preferred_name_space_uri)
+        rewrite_condition = if preferred_name_space_uri
+                              "RewriteCond %{REQUEST_URI} ^.*#{URI.parse(preferred_name_space_uri).path[1..-1]}.*$"
+                            elsif ontology_uri
+                              "RewriteCond %{REQUEST_URI} ^.*#{URI.parse(ontology_uri).path[1..-1]}.*$"
+                            end
+      
+        ontology_rule = if ontology_uri
+                          ontology_uri += '/' unless ontology_uri.end_with?('/')
+                          "RewriteRule ^#{URI.parse(ontology_uri).path[1..-1]}?$ #{ontology_portal_url} [R=301,L]"
+                        end
+      
         <<-HTACCESS.strip_heredoc
-            RewriteEngine On
-            #{ontology_rule if ontology_rule}
-            RewriteRule ^.*/([^/#]+)$ #{ontology_portal_url}/$1 [R=301,L]
+          RewriteEngine On
+          #{ontology_rule if ontology_rule}
+          #{rewrite_condition if rewrite_condition}
+          RewriteRule ^.*/([^/#]+)/?$ #{ontology_portal_url}/$1 [R=301,L]
         HTACCESS
     end
       
-    def generate_nginx_content(ontology_portal_url, ontology_uri)
+    def generate_nginx_content(ontology_portal_url, ontology_uri, preferred_name_space_uri)
         ontology_rule = nil
+        rewrite_condition = if preferred_name_space_uri
+            URI.parse(preferred_name_space_uri).path[1..-1]
+          elsif ontology_uri
+            ontology_uri += '/' unless ontology_uri.end_with?('/')
```

Comment:
why this ?

Reply:
why the `+='/'` ?

Reply:
this is because there are some ontology URIs that end with `/` and others don't end with `/`, so i add it to handle the redirection for both cases `/path/resource_id` and `/path/resource_id/`

example 1: 
- URI = `http://domain/path/ontology1/` => redirection rule will be `^path/ontology1/$` => this rule will not redirect `path/ontology1` (without `/` in the end because it expect to end with /)

example 2:
- URI = `http://domain/path/ontology2/` => redirection rule will be `^path/ontology2$` => this rule will not redirect `^path/ontology2/$` (with `/` in the end because it expect to end with letter 2)

Solution
- we add the `/` in the end of the URI, so all uris will end with slash
- redirection rule will be : `^path/ontology/?$` in this redirection rule i have added `/?` in the end to handle request for path/ontology and path/ontology/